docs/source/newsfragments/4098.feature.1.rst

```diff
@@ -0,0 +1 @@
+:class:`~cocotb.types.LogicArray` can take :class:`int` for the ``range`` argument as shorthand for ``Range(range-1, "downto", 0)``.
```

Comment:
Do we really need to overload the `range` argument here? `range=4` reads strange, more common naming would be `size=4` or `width=4`. How about keeping range as-is, and adding a new named argument (e.g., `size` or `width`) that takes an integer?

Reply:
I thought about this. It might make the constructor slightly more complicated, making things slower, but I guess since we are using the trusted constructor for handles this is fine since only manually constructed LogicArrays would hit it.

I'll do that.

Reply:
This change actually shuffles a lot of code. I need to come up with a better scheme than I have now.

Reply:
I punted on that change for another PR.